src/helpers/dappNameHandler.js

```diff
@@ -1,9 +1,143 @@
-  'Aave - Open Source Protocol for Money markets': 'Aave',
-  'Pool - PoolTogether - App': 'PoolTogether',
+  '1inch.exchange': {
+    name: '1inch',
+    uri: null,
+  },
+  '88mph.app': {
+    name: '88mph',
+    uri: buildAssetUrl('88mph.app'),
+  },
+  'aave.com': {
+    name: 'Aave',
+    uri: buildAssetUrl('aave.com'),
+  },
+  'balancer.exchange': {
+    name: 'Balancer',
+    uri: null,
+  },
+  'compound.finance': {
+    name: 'Compound',
+    uri: buildAssetUrl('compound.finance'),
+  },
+  'cream.finance': {
+    name: 'Cream',
+    uri: buildAssetUrl('cream.finance'),
+  },
+  'curve.fi': {
+    name: 'Curve',
+    uri: null,
+  },
+  'defisaver.com': {
+    name: 'DeFi Saver',
+    uri: buildAssetUrl('defisaver.com'),
+  },
+  'dydx.exchange': {
+    name: 'dYdX',
+    uri: buildAssetUrl('dydx.exchange'),
+  },
+  'instadapp.io': {
+    name: 'Instadapp',
+    uri: null,
+  },
+  'kyberswap.com': {
+    name: 'KyberSwap',
+    uri: buildAssetUrl('kyberswap.com'),
+  },
+  'matcha.xyz': {
+    name: 'Matcha',
+    uri: buildAssetUrl('matcha.xyz'),
+  },
+  'mstable.org': {
+    name: 'mStable',
+    uri: buildAssetUrl('mstable.org'),
+  },
+  'oasis.app': {
+    name: 'Oasis',
+    uri: null,
+  },
+  'opensea.io': {
+    name: 'OpenSea',
+    uri: null,
+  },
+  'piedao.org': {
+    name: 'PieDAO',
+    uri: null,
+  },
+  'pooltogether.com': {
+    name: 'PoolTogether',
+    uri: buildAssetUrl('pooltogether.com'),
+  },
+  'rarible.com': {
+    name: 'Rarible',
+    uri: buildAssetUrl('rarible.com'),
+  },
+  'sushiswapclassic.org': {
+    name: 'SushiSwap',
+    uri: buildAssetUrl('sushiswapclassic.org'),
+  },
+  'swerve.fi': {
+    name: 'Swerve',
+    uri: buildAssetUrl('swerve.fi'),
+  },
+  'synthetix.exchange': {
+    name: 'Synthetix',
+    uri: buildAssetUrl('synthetix.exchange'),
+  },
+  'umaproject.org': {
+    name: 'UMA',
+    uri: null,
+  },
+  'uniswap.org': {
+    name: 'Uniswap',
+    uri: null,
+  },
+  'walletconnect.org': {
+    name: 'WalletConnect',
+    uri: null,
+  },
+  'yam.finance': {
+    name: 'YAM',
+    uri: buildAssetUrl('yam.finance'),
+  },
+  'yearn.finance': {
+    name: 'yearn',
+    uri: null,
+  },
+  'zapper.fi': {
+    name: 'Zapper',
+    uri: buildAssetUrl('zapper.fi'),
+  },
+  'zerion.io': {
+    name: 'Zerion',
+    uri: null,
+  },
 });
 
-export const convertDappNameToDisplay = name => {
-  return DisplayDappNames[name] || name;
+export const dappNameOverride = url => {
+  const hostname = getDappHostname(url);
+  return DisplayDappNames[hostname]?.name;
+};
+
+export const dappLogoOverride = url => {
+  const hostname = getDappHostname(url);
+  return DisplayDappNames[hostname]?.uri;
+};
+
+export const getDappHostname = url => {
+  const urlObject = new URL(url);
+  let hostname;
+  const subdomains = urlObject.hostname.split('.');
+  if (subdomains.length === 2) {
+    hostname = urlObject.hostname;
+  } else {
+    hostname = `${subdomains[subdomains.length - 2]}.${
```

Comment:
If I understand this correctly, for any hostnames with greater than 2 subdomains, we would just take the second to last and last subdomain. So does this mean for hostnames such as "myetherwallet.com.ua" (from the cryptoscamdb list, there are many like it), it would just end up being "com.ua"?

Reply:
Perhaps we should look at some js tld / fld libs

Reply:
Good call!  I was supposed to pass the entire host, not just the top level 😅 
Just fixed that and moved the logic to the approval sheet, so we check when you connect to the dapp instead of on every transaction.